mm2src/mm2_main/src/lp_ordermatch/simple_market_maker.rs

```diff
@@ -614,81 +639,69 @@ async fn execute_create_single_order(
 
 async fn process_bot_logic(ctx: &MmArc) {
     let simple_market_maker_bot_ctx = TradingBotContext::from_ctx(ctx).unwrap();
-    let state = simple_market_maker_bot_ctx.trading_bot_states.lock().await;
-    let (cfg, price_url) = if let TradingBotState::Running(running_state) = &*state {
-        let res = (running_state.trading_bot_cfg.clone(), running_state.price_url.clone());
-        drop(state);
-        res
-    } else {
-        drop(state);
-        return;
+    let mut state = simple_market_maker_bot_ctx.trading_bot_states.lock().await;
+    let running_state = match &mut *state {
+        TradingBotState::Running(running_state) => running_state,
+        TradingBotState::Stopping(_) | TradingBotState::Stopped(_) => return,
     };
-    let rates_registry = match fetch_price_tickers(price_url.as_str()).await {
-        Ok(model) => {
-            info!("price successfully fetched from {price_url}");
-            model
-        },
+
+    let cfg = running_state.trading_bot_cfg.clone();
+    let price_urls = &mut running_state.price_urls;
+    let rates_registry = match fetch_price_tickers(price_urls).await {
+        Ok(model) => model,
         Err(err) => {
             let nb_orders = cancel_pending_orders(ctx, &cfg).await;
             error!("error fetching price: {err:?} - cancel {nb_orders} orders");
             return;
         },
     };
 
+    drop(state);
```

Comment:
you can also use this to drop a value 👁️ 
```rust
let _ = state
```

Reply:
I would prefer to leave the explicit drop :)